akka-http-core/src/main/scala/akka/http/scaladsl/util/FastFuture.scala

```diff
@@ -28,7 +28,7 @@ class FastFuture[A](val future: Future[A]) extends AnyVal {
   def filter(pred: A ⇒ Boolean)(implicit executor: ExecutionContext): Future[A] =
     flatMap { r ⇒
       if (pred(r)) future
-      else throw new NoSuchElementException("Future.filter predicate is not satisfied") // FIXME: avoid stack trace generation
+      else throw new NoSuchElementException("Future.filter predicate is not satisfied")
```

Comment:
You can use the Scala's built-in [`NoStackTrace`](https://github.com/scala/scala/blob/v2.12.3/src/library/scala/util/control/NoStackTrace.scala#L22), so this becomes `new NoSuchElementException(...) with NoStackTrace`.